modules/targetVideoBidAdapter.js

```diff
@@ -0,0 +1,179 @@
+    const tags = bidRequests.map(createVideoTag);
+    const schain = bidRequests[0].schain;
+    const payload = {
+      tags: [...tags],
+      sdk: {
+        source: SOURCE,
+        version: '$prebid.version$'
+      },
+      schain: schain
+    };
+    return formatRequest(payload, bidderRequest);
+  },
+
+  /**
+   * Unpack the response from the server into a list of bids.
+   *
+   * @param {*} serverResponse A successful response from the server.
+   * @return {Bid[]} An array of bids which were nested inside the server.
+   */
+  interpretResponse: function(serverResponse, { bidderRequest }) {
+    serverResponse = serverResponse.body;
+    const bids = [];
+
+    if (serverResponse.tags) {
+      serverResponse.tags.forEach(serverBid => {
+        const rtbBid = getRtbBid(serverBid);
+        if (rtbBid && rtbBid.cpm !== 0 && rtbBid.ad_type == VIDEO) {
+          bids.push(newBid(serverBid, rtbBid, bidderRequest));
+        }
+      });
+    }
+    return bids;
+  }
+
+}
+
+function getSizes(request) {
+  let sizes = request.sizes;
+  if (request.mediaTypes && request.mediaTypes.banner && request.mediaTypes.banner.sizes) {
+    sizes = request.mediaTypes.banner.sizes;
+  }
+  return sizes;
+}
+
+function formatRequest(payload, bidderRequest) {
+  const options = {
+    withCredentials: true
+  };
+  const request = {
+    method: 'POST',
+    url: ENDPOINT_URL,
+    data: JSON.stringify(payload),
+    bidderRequest,
+    options
+  };
+
+  return request;
+}
+
+/**
+ * Create video auction.
+ *
+ * @param {*} serverResponse A successful response from the server.
+ * @return {Bid[]} An array of bids which were nested inside the server.
+ */
+function createVideoTag(bid) {
+  const tag = {};
+  tag.id = parseInt(bid.params.placementId, 10);
+  tag.gpid = 'targetVideo';
+  tag.sizes = getSizes(bid);
+  tag.primary_size = tag.sizes[0];
+  tag.ad_types = [VIDEO];
+  tag.uuid = bid.bidId;
+  tag.allow_smaller_sizes = false;
+  tag.use_pmt_rule = false
+  tag.prebid = true;
+  tag.disable_psa = true;
+  tag.hb_source = 1;
+  tag.require_asset_url = true;
+  tag.video = {
+    playback_method: 2,
+    skippable: true
+  };
+
+  return tag;
+}
+
+/**
+ * Unpack the Server's Bid into a Prebid-compatible one.
+ * @param serverBid
+ * @param rtbBid
+ * @param bidderRequest
+ * @return Bid
+ */
+function newBid(serverBid, rtbBid, bidderRequest) {
+  const bidRequest = getBidRequest(serverBid.uuid, [bidderRequest]);
+  const bid = {
+    requestId: serverBid.uuid,
+    cpm: rtbBid.cpm * MARGIN,
+    creativeId: rtbBid.creative_id,
+    dealId: rtbBid.deal_id,
+    currency: 'USD',
+    netRevenue: true,
+    width: bidRequest.sizes[0][0],
+    height: bidRequest.sizes[0][1],
+    ttl: 300,
+    adUnitCode: bidRequest.adUnitCode,
+    appnexus: {
+      buyerMemberId: rtbBid.buyer_member_id,
+      dealPriority: rtbBid.deal_priority,
+      dealCode: rtbBid.deal_code
+    }
+  };
+
+  if (rtbBid.rtb.video) {
+    Object.assign(bid, {
```

Comment:
Can lines 152 and 153 be moved within this `Object.assign`? Otherwise, can lines 148 and 149 be changed to `bid.<property> = <value>`?

Reply:
You're right, moved within Object.assign.